core/src/Cuda/Kokkos_Cuda_View.hpp

```diff
@@ -160,7 +160,7 @@ struct CudaLDGFetch {
   CudaLDGFetch(const CudaLDGFetch& rhs) : m_ptr(rhs.m_ptr) {}
 
   KOKKOS_INLINE_FUNCTION
-  CudaLDGFetch(CudaLDGFetch&& rhs) : m_ptr(rhs.m_ptr) {}
+  CudaLDGFetch(CudaLDGFetch&& rhs) noexcept : m_ptr(rhs.m_ptr) {}
```

Comment:
depending on compiler this now injects catch clauses into performance crticial code.